[Webster Tan Jia Sheng] iP

webtjs opened the pull request.

webtjs committed:
Rename chatbot and refactor files

webtjs committed:
Helpy.java: Add greeting and exit

webtjs committed:
Add echo command feature, update greeting and exit message

webtjs committed:
Add Task class, add task storage and retrieval

webtjs committed:
Add mark and unmark task feature

Add new member isDone to Task class to keep track of task completion.

Add two new methods markTask and unmarkTask to update the status of the
task.

webtjs committed:
Helpy.java: Modify to comply with the given coding standard

webtjs committed:
Helpy.java: Add constant and refactor code

cyhjason29 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+        return isDone;
+    }
+
+    public void setDone(boolean done) {
```
Maybe the boolean variable 'done' can be made to sound more boolean? eg 'isMarked'

cyhjason29 commented on src/main/java/Helpy.java:
```diff
@@ -0,0 +1,123 @@
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void goodbyeUser() {
```
Maybe use a verb in the name to represent this method instead? eg 'sayGoodbye'

cyhjason29 commented on src/main/java/Helpy.java:
```diff
@@ -0,0 +1,123 @@
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void main(String[] args) {
+        greetUser();
+
+        Scanner in = new Scanner(System.in);
+        String command = "";
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        while (!command.equals("bye")) {
+            command = in.nextLine();
+            switch (command) {
+            case "bye":
+                goodbyeUser();
+                break;
+            case "list":
+                listTasks(taskList);
+                break;
+            default:
+                addTask(command, taskList);
+                break;
+            }
+        }
+    }
```
I like how clean this section of code is; Appropriate indentations and layout help to make it more readable.

cyhjason29 commented on src/main/java/Helpy.java:
```diff
@@ -0,0 +1,123 @@
+import java.util.Scanner;
+import java.util.ArrayList;
```
I like how you explicitly listed out the imported classes.

cyhjason29 reviewed (commented):
Overall, the layout, statements and comments follow the coding standards very well. Apart from some minor improvements required in the naming, the code is readable and easy to follow.

webtjs committed:
Add Todo, Deadline, and Event classes, Helpy.java: Refactor code

Add new method addMessage in Helpy.java to be used whenever a new todo,
deadline, or event is added.

webtjs committed:
Implement Automated Text UI Testing

webtjs committed:
Helpy.java: Add error message for invalid command

webtjs committed:
Helpy.java: Add error handling for mark and unmark commands

webtjs committed:
Helpy.java: Refactor code, combine mark and unmark task together

webtjs committed:
Add error handling when creating todo, deadline and event objects

Define new exception IllegalDescriptionException to catch empty
todo description

webtjs committed:
Helpy.java: Refactor code by replacing repeated code with a method

webtjs committed:
Merge branch 'branch-Level-5'

webtjs committed:
Organise classes into packages

Create helpy package, exceptions package, and task package

webtjs committed:
Merge branch 'branch-A-Packages'

Jonoans commented on src/main/java/helpy/Helpy.java:
```diff
@@ -0,0 +1,157 @@
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void processCommand(String command, ArrayList<Task> taskList) {
```
The processCommand method is rather long, try to abstract out the processing implementations of each command.

Jonoans commented on src/main/java/helpy/exceptions/IllegalDescriptionException.java:
```diff
@@ -0,0 +1,4 @@
+package helpy.exceptions;
+
+public class IllegalDescriptionException extends Exception{
```
Missing space here

Jonoans commented on src/main/java/helpy/task/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+        super();
+        String description = command.replace("deadline", "").trim();
+        String[] details = description.split("/by");
+        this.taskName = details[0].trim();
+        this.dueDate = details[1].trim();
```
Same here, try to avoid magic numbers

Jonoans commented on src/main/java/helpy/task/Event.java:
```diff
@@ -0,0 +1,23 @@
+        super();
+        String description = command.replace("event", "").trim();
+        String[] details = description.split("/from");
+        taskName = details[0].trim();
+
+        String[] eventPeriod = details[1].split("/to");
+        eventFrom = eventPeriod[0].trim();
+        eventTo = eventPeriod[1].trim();
```
Avoid magic numbers like this, from a reader's perspective, might not be immediately apparent what these numbers mean

Jonoans reviewed (commented):
Decent job overall, though do try to use functions to abstract out processing details, this helps to make your code units smaller

webtjs committed:
Add delete task function

webtjs committed:
Add feature to load task data when chatbot starts up

webtjs committed:
Add feature to save new tasks into a .txt file

webtjs committed:
Add feature to update save data whenever task list changes

webtjs committed:
Merge branch 'branch-Level-6'

webtjs committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/helpy/Helpy.java

webtjs committed:
Update save data whenever a task is removed from the list

webtjs committed:
Helpy.java: Add save file creation when no previous save file exists

webtjs committed:
Add Ui Class for interactions with user

webtjs committed:
Refactor code to make the code more OOP

Create Ui class for interaction with user
Create Storage class to load and save data
Create Parser class to interpret user command
Create TaskList class containing operations for the task list

Create *Command classes for the various command types

webtjs committed:
Add feature for users to find a task using keywords

Create FindCommand class to search for tasks containing a keyword

webtjs committed:
Add date recognition feature for deadline and event tasks

Create EventDateSequenceException to deal with event dates not in
chronological order

webtjs committed:
Add JavaDoc comments to the code

webtjs committed:
Merge pull request #2 from webtjs/branch-Level-8

Add date recognition feature for deadline and event tasks

webtjs committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/helpy/Ui.java

webtjs committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/helpy/Parser.java
#	src/main/java/helpy/Storage.java
#	src/main/java/helpy/command/EventCommand.java

webtjs committed:
Merge pull request #3 from webtjs/branch-Level-9

Add feature for users to find a task using keywords

webtjs committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/helpy/Ui.java
#	src/main/java/helpy/task/TaskList.java

webtjs committed:
Merge pull request #1 from webtjs/branch-A-JavaDoc

Add JavaDoc comments to the code

webtjs committed:
Add help feature to display all commands for the chatbot

Create HelpCommand class to execute the help command

webtjs committed:
Add a User Guide for the project

webtjs committed:
docs/README.md: Fix text alignment

webtjs committed:
Add additional JavaDoc comments

webtjs committed:
Ui.java: Add a message when trying to list an empty task list

webtjs committed:
Ui.java: Change help message

webtjs committed:
docs/README.md: Add additional segment for find command under appendix

webtjs committed:
Create _config.yml

Add a theme to GitHub Page
